arches/app/views/resource.py

```diff
@@ -653,7 +653,8 @@ def get(
             permitted_edits = []
             for edit in edits:
                 if edit.nodegroupid is not None:
-                    if request.user.has_perm("read_nodegroup", edit.nodegroupid):
+                    edit_nodegroup = models.NodeGroup.objects.get(pk=edit.nodegroupid)
```

Comment:
Noticed we have a query in a loop. We don't have an FK from EditLog to NodeGroup, so we can't use select_related() (without hacks like #11809), but we could still avoid N+1 queries by pulling the nodegroup query out of the loop.

```py
nodegroup_ids_from_edits = [
    UUID(ng) if ng else ng # this could be avoided by fixing the edits query to exclude Nones
    for ng in edits.values_list('nodegroupid', flat=True)
]
nodegroups_for_edits = (
    models.NodeGroup.objects
    .filter(pk__in=nodegroups_ids_from_edits)
    .in_bulk()
)
```
`in_bulk()` is just shorthand for making a queryset into a cache:`{obj.pk: obj for obj in query}` so you can look up an instance by pk. The explicit form is fine, too.